pkg/proxy/iptables/proxier.go

```diff
@@ -569,6 +571,40 @@ func (proxier *Proxier) OnEndpointsDelete(endpoints *v1.Endpoints) {
 func (proxier *Proxier) OnEndpointsSynced() {
 	proxier.mu.Lock()
 	proxier.endpointsSynced = true
+	proxier.setInitialized(proxier.servicesSynced && proxier.endpointSliceSynced)
+	proxier.mu.Unlock()
+
+	// Sync unconditionally - this is called once per lifetime.
+	proxier.syncProxyRules()
+}
+
+// CacheEndpointSlices loads Endpoint Slices into the cache
+func (proxier *Proxier) CacheEndpointSlices(endpointSlices []*discovery.EndpointSlice) {
+	proxier.endpointsChanges.CacheEndpointSlices(endpointSlices)
+	proxier.OnEndpointSliceSynced()
+}
+
+// OnEndpointSliceUpdate is called whenever an endpoint slice creation or
+// deletion is observed.
+func (proxier *Proxier) OnEndpointSliceUpdate(endpointSlice *discovery.EndpointSlice) {
```

Comment:
consider this: 

```

type syncFunc func() 

type EndpointSliceTrackerAdapter struct {
      tracker EndpointChangeTracker
      sync syncFunc
}

func (*EndpointSliceTrackerAdapter a) OnEndpointSliceAdd ...
func (*EndpointSliceTrackerAdapter a) OnEndpointSliceUpdate ...
func (*EndpointSliceTrackerAdapter a) OnEndpointSliceDelete ...


func NewEndpointSliceTrackerAdapter(tracker EndpointChangeTracker, sync syncFunc) EndpointSliceHandler 


// In iptables/ipvs proxier
type Proxier struct {
	EndpointSliceTrackerAdapter
...



func NewIptablesProxier(...) *Proxier {

	tracker := NewEndpointChangeTracker()
	p := &Proxier{...}
	...

	adapter := NewEndpointSliceTrackerAdapter(tracker, p.Sync)

	p.EndpointSliceTrackerAdapter = adapter

	return p

}
```

Reply:
After discussing this further we decided that this would become more involved than desired here, but still worth doing in the future to help clean up all the duplicate code between proxiers.